app/models/campaign.rb

```diff
@@ -0,0 +1,63 @@
+class Campaign < ApplicationRecord
+  include AASM
+
+  has_many :campaign_templates
+  has_many :templates, through: :campaign_templates
+  has_many :campaign_contacts
+  has_many :contacts, through: :campaign_contacts
+
+  validates :name, presence: { message: 'Campaign name is required' }
+  validates :campaign_type, presence: { message: 'Campaign type is required' }
+  validates :email_limit, presence: { message: 'Email limit is required' }
+  validates :start_time, presence: { message: 'Start time is required' }
+  validates :end_time, presence: { message: 'End time is required' }
+  validates :campaign_run_time, presence: { message: 'Campaign run time is required' }
+  validates :batch_contact, presence: { message: 'Campaign batch contact is required' }
+
+  after_commit :schedule_email_job
+
+  aasm column: 'status' do
+    state :draft, initial: true
+    state :saved
+    state :pause
+    state :disabled
+
+    event :save_campaign do
+      transitions from: :draft, to: :saved
+    end
+
+    event :pause_campaign do
+      transitions from: :saved, to: :pause
+    end
+
+    event :resume_campaign do
+      transitions from: :pause, to: :saved
+    end
+
+    event :disable_campaign do
+      transitions from: [:saved, :pause], to: :disabled
+    end
+  end
+
+  private
+
+  def schedule_email_job
```

Comment:
As per our discussion, Will it send the subsiquent template after certain number of days from the first template and so on ?